### DISCUSSION TITLE
how to find out the secret key

### DISCUSSION THREAD

**Discussion by NeilRiver** [keycloak/keycloak#24118] [Q&A] [answered]
how to find out the secret key

data = {
    "client_id": "formio-kolya",
    "grant_type": "password",
    "username": "kolya",
    "password": "kolya",
}

response = requests.post(
    "xxx/realms/Formio/protocol/openid-connect/token",
    data=data,
    verify=False,
).json()

token = response["access_token"]

decoded_token = jwt.decode(
    token,
    ???,    **<----------- how to find out the secret key**
    "HS256",
)

**Comment by jonkoops** [comment]
You can get the public key from Keycloak's REST API. I recommend reading the discussion under https://github.com/keycloak/keycloak-nodejs-connect/issues/492

**Reply by NeilRiver** [reply]
my goal is to pass the access token to formio and verify it

**Reply by NeilRiver** [reply, marked as answer]
1) get token 
http://srv-apps-pre-1.main.velcom.by:8080/realms/Formio/protocol/openid-connect/token

2) verify token, in settings:  Client authentication -> on , Direct access grants -> on
http://srv-apps-pre-1.main.velcom.by:8080/realms/Formio/protocol/openid-connect/token/introspect
{
token,
client_id,
client_secret
}

>>  {
    "active": false
}

this is what I wanted
